poll/poll_linux.go

```diff
@@ -115,6 +118,8 @@ func (p *Poller) WaitWriteCtx(ctx context.Context) error {
 		return nil
```

Comment:
Epoll shouldn't return if it wasn't one of those two, it should either return one of those two cases or error out on the EpollWait call.